### DISCUSSION TITLE
Frequency of Triple alpha variant clustering for Titanium

### DISCUSSION THREAD

**Discussion by Rhalder322** [ORTools4MTEX/ORTools#71] [Q&A] [answered]
As we know that the transformation strains are minimized by self-accommodation mechanisms via triple alpha variants clustering in Ti64. Generally, Cluster I represents A/B/C, D/E/F, G/H/I, J/K/L; while Category II encompasses those that consist of variants A/E/I, B/D/L, C/G/K and F/H/J. 

Is there any feature available within MTEX to estimate the number or area frequency of triple alpha variant boundaries?

**Comment by frankNiessen** [comment]
Hi @Rhalder322 ,

the answer is yes and no. There is no built-in turn-key solution, but it is rather straight forward to program what you are looking for with few lines of code. Assuming you have run [Example 8](https://github.com/ORTools4MTEX/ORTools/blob/develop/ORTools_example08.m) to the end you should have a reconstructed prior beta microstructure and a variant ID assigned to each transformed alpha grain. I don't know exactly which convention you use to define variants A/B/C etc, but the convention used in ORTools for variants 1-12 is plotted into the command window during execution of example 8. This code snippet will get you the variant Ids at each triple junction that is made up of transformed alpha grains:

```
% Get the triple points of all transformed alpha grains:
triplePoints = job.transformedGrains.triplePoints;
% Check that all grains at triple junction are transformed grains
ind = all(ismember(triplePoints.grainId, job.transformedGrains.id), 2)
triplePointGrains = job.transformedGrains(job.transformedGrains.id2ind(triplePoints(ind).grainId))
% Get the variant Ids
triplePointVariantIds = triplePointGrains.variantId
% Potentially sort them
sort(triplePointVariantIds,2)
```

Best wishes
Frank

**Reply by Rhalder322** [reply]
Hello Prof. Niessen,

The histogram plot of hist(triplePointVariantIds) provides me with the following plot:

![triple alpha variants histogram](https://github.com/ORTools4MTEX/ORTools/assets/158861570/1d1538cb-dbf1-46e7-8082-3b1421bb5916)

**_

### How do I figure out which triple alpha variants groups are which from this plot? 

### Moreover, I would like to define variant clusters and plot the number & area frequencies of them. How do I do that?

_**

% Define the groups of triple variant clusters
vGroupIds = {
    [1 2 3],...
    [4 5 6],...
    [7 8 9],...
    [10 11 12],...
    [1 11 9],...
    [2 12 6],...
    [3 8 5],...
    [10 7 4]
    };


### _**Secondly, how do we relate variants used in these two conventions mentioned in the image below?**_

![alpha variants conventions](https://github.com/ORTools4MTEX/ORTools/assets/158861570/b83a65d6-23a7-4d6f-b28d-4dae2cbd4126)

Thank you for helping me with this.

Regards,
Rajib Halder

**Reply by frankNiessen** [reply]
Hi, 

so all challenges you have from here on mostly not MTEX/ORTools challenges but MATLAB challenges. 

Your first questions is essentially: How often is each row of `vGroupIds` represented in the array `triplePointVariantIds`, this is an easy task for ChatGPT these days. To compute this it is essential to sort the values of each row first. Then you also want to take into account how large each cluster is to not only count the number of occurences, but the area of each cluster.  

The entire code looks like this.

```
% Get the triple points of all transformed alpha grains:
triplePoints = job.transformedGrains.triplePoints;
% Check that all grains at triple junction are transformed grains
ind = all(ismember(triplePoints.grainId, job.transformedGrains.id), 2);
triplePointGrains = job.transformedGrains(job.transformedGrains.id2ind(triplePoints(ind).grainId));

% Get the variant Ids
triplePointVariantIds = sort(triplePointGrains.variantId,2);
% Get the number of EBSD pixels per triple point grain cluster
clusterSz = sum(triplePointGrains.grainSize,2);
% Define the variantId groups
vGroupIds = sort([1 2 3;...
4 5 6;...
7 8 9;...
10 11 12;...
1 11 9;...
2 12 6;...
3 8 5;...
10 7 4],2);

% Check how many rows in triplePointVariantIds match the different vGroups
% and consider the area of each occurence
for i = 1:size(vGroupIds, 1)
    globalClusterSz(i) = sum(ismember(triplePointVariantIds, vGroupIds(i, :), 'rows').*clusterSz);
end
```

The numbers you get in `globalClusterSz` give you the number of EBSD pixels in your map that belong to each cluster.


Regarding the second question, if you run the latest ORTools version, the command `ORinfo(job.p2c);` will return this information. Alternatively you can type `job.p2c.variants(1)` into the command window to get this information. 

Best wishes
Frank

**Comment by Rhalder322** [comment]
Hello Prof. Niessen,

Q1. For a specific prior beta grain, all 12 alpha variants ( V1, V2, V3.........V12) can be identified based on the particular orientation (P1) of the beta grain and considering the Burgers orientation relationship. For another beta grain (P1*) having a different orientation than the previous one, the orientation of all 12 alpha variants  ( V1*, V2*, V3*.........V12*) should also be different than the previous one. Therefore, how a specific alpha variant (V1, V2, V3.........V12) is identified in an EBSD map having several different orientation-based beta grains?

Q2.  Is it possible to get information of the corresponding parent phase plane & direction for a specific alpha variant defined by unique axis and rot. axis? For example, for the variants listed below, how to figure out which (111) plane and [110] direction is related to which alpha variant?

![image](https://github.com/ORTools4MTEX/ORTools/assets/158861570/653525c0-fc34-42aa-ac22-8b1d9220d261)

Thank you for your support.

Regards,
Rajib Halder

**Reply by AzdiarGazder** [reply, marked as answer]
Hi @Rhalder322,

Here are the answers to your questions in order:
1. Please have a read of the [MTEX help section on how child variant orientations are calculated for given a parent orientation](https://mtex-toolbox.github.io/orientation.variants.html).

2. As Frank stated in his earlier reply to you:
"Regarding the second question, if you run the latest ORTools version, the command `ORinfo(job.p2c);` will return this information. Alternatively you can type `job.p2c.variants(1)` into the command window to get this information."


<ins>**Lastly, and importantly, if an explanation from the maintainers resolves your issue, please press the "Mark as answer" button on the appropriate response. Your assistance in this regard enables us to close discussions as complete. **</ins>

Hope this helps.

Warm regards,
Azdi
